File: subject-service/src/main/kotlin/dev/arkaan/schoolapp/subjectservice/db/entity/Subject.kt
package dev.arkaan.schoolapp.subjectservice.db.entity

import org.jetbrains.exposed.sql.Column
import org.jetbrains.exposed.sql.Table

object Subject: Table("subject") {
    private val id: Column<UShort> = ushort("id").autoIncrement()
    val subjectCode: Column<String> = varchar("subject_code", 20)
    val name: Column<String> = varchar("name", 100)
    val description: Column<String> = text("description")

    override val primaryKey = PrimaryKey(id)
}

File: backoffice/lib/features/subject/subject.dart
class Subject {
  final int? id;
  String subjectCode;
  String name;
  String description;

  Subject(this.id, this.subjectCode, this.name, this.description);

  factory Subject.empty() => Subject(-1, "", "", "");

  Subject.fromJson(Map<String, dynamic> json)
    : id = json["id"],
      subjectCode = json["subjectCode"],
      name = json["name"],
      description = json["description"];
  
  Map<String, dynamic> toJson() => {
    "id": id,
    "subjectCode": subjectCode,
    "name": name,
    "description": description
  };
}